Sources/BluemixAppID/WebAppKituraCredentialsPlugin.swift

```diff
@@ -144,14 +136,13 @@ public class WebAppKituraCredentialsPlugin: CredentialsPluginProtocol {
         var authUrl = generateAuthorizationUrl(options: options)
         
         // If there's an existing anonymous access token on session - add it to the request url
-        let appIdAuthContext = request.session?[WebAppKituraCredentialsPlugin.AuthContext] as? [String : Any]
-        if let context = appIdAuthContext, let payload = context["accessTokenPayload"] as? SwiftyJSON.JSON, payload["amr"][0] == "appid_anon" {
-            logger.debug("WebAppKituraCredentialsPlugin :: handleAuthorization :: added anonymous access_token to url")
-            authUrl += "&appid_access_token=" + ((context["accessToken"] as? String) ?? "")
-        }
-        
-        // If previous anonymous access token not found and new anonymous users are not allowed - fail
-        if appIdAuthContext  == nil && allowAnonymousLogin == true && allowCreateNewAnonymousUser != true {
+        if let appIdAuthContext = request.session?[WebAppKituraCredentialsPlugin.AuthContext] as? [String : Any] {
+            if ((appIdAuthContext["accessTokenPayload"] as? [String : Any])?["amr"] as? [String])?[0] ==  "appid_anon" {
```

Comment:
This is really hard to read. Can you break it up to make it more readable? like:
```
let payload = appIdAuthContext["accessTokenPayload"] as? [String : Any]
if ( payload?["amr"] as? [String])? [0]...
```